src/index/index.c

```diff
@@ -0,0 +1,214 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Apache License, Version 2.0,
+* modified with the Commons Clause restriction.
+*/
+
+#include "index.h"
+#include "index_type.h"
+
+RedisModuleKey* _index_LookupKey(RedisModuleCtx *ctx, const char *graph, const char *label, const char *property) {
+  char *strKey;
+  int keylen = asprintf(&strKey, "%s_%s_%s_%s", INDEX_PREFIX, graph, label, property);
+
+  RedisModuleString *rmIndexId = RedisModule_CreateString(ctx, strKey, keylen);
+  free(strKey);
+
+  RedisModuleKey *key = RedisModule_OpenKey(ctx, rmIndexId, REDISMODULE_WRITE);
+  RedisModule_FreeString(ctx, rmIndexId);
+
+  return key;
+}
+
+/* Memory management and comparator functions that get attached to
+ * string and numeric skiplists as function pointers. */
+int compareNodes(GrB_Index a, GrB_Index b) {
+  return a - b;
+}
+
+int compareStrings(SIValue *a, SIValue *b) {
+  return strcmp(a->stringval, b->stringval);
+}
+
+int compareNumerics(SIValue *a, SIValue *b) {
+  return a->doubleval - b->doubleval;
+}
+
+/* The index must maintain its own copy of the indexed SIValue
+ * so that it becomes outdated but not broken by updates to the property. */
+void cloneKey(SIValue **property) {
+  SIValue *clone = *property;
+  *clone = SI_Clone(*clone);
+}
+
+void freeKey(SIValue *key) {
+  SIValue_Free(key);
+}
+
+/* Construct key and retrieve index from Redis keyspace */
+Index* Index_Get(RedisModuleCtx *ctx, const char *graph, const char *label, const char *property) {
+  Index *idx = NULL;
+  char *strKey;
+  int keylen = asprintf(&strKey, "%s_%s_%s_%s", INDEX_PREFIX, graph, label, property);
+
+  RedisModuleString *rmIndexId = RedisModule_CreateString(ctx, strKey, keylen);
+  free(strKey);
+
+  RedisModuleKey *key = RedisModule_OpenKey(ctx, rmIndexId, REDISMODULE_READ);
+  RedisModule_FreeString(ctx, rmIndexId);
+
+  if (RedisModule_ModuleTypeGetType(key) == IndexRedisModuleType) {
+    idx = RedisModule_ModuleTypeGetValue(key);
+  }
+
+  RedisModule_CloseKey(key);
+
+  return idx;
+}
+
+void Index_Delete(RedisModuleCtx *ctx, const char *graphName, const char *label, const char *prop) {
+  RedisModule_ReplyWithArray(ctx, 1);
+  RedisModule_ReplyWithArray(ctx, 2);
```

Comment:
It should be clear for each reply array, where its content is coming from,
either add a comment explaining this, or if possible move reply "declaration" to be in the same scope as the actual reply data.
Please explain reply structure